micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -34,27 +35,38 @@ pub enum NVSStorageError {
     NVSValueDecodeError(#[from] DecodeError),
     #[error(transparent)]
     NVSUriParseError(#[from] InvalidUri),
+    #[error("nvs collection empty")]
+    NVSCollectionEmpty(#[from] EmptyStorageCollectionError),
 }
 
 #[derive(Clone)]
 pub struct NVSStorage {
     // esp-idf-svc partition driver ensures that only one handle of a type can be created
     // so inner mutability can be achieves safely with RefCell
     nvs: Rc<RefCell<EspCustomNvs>>,
-    partition_name: CString,
```

Comment:
store a copy of the partition rather than the partition name to avoid cloning a CString as we pass the Storage through different par to of the code